frontend/src/components/search/__tests__/AdvancedSearch.test.tsx

```diff
@@ -0,0 +1,213 @@
+import '@testing-library/jest-dom';
+import { render, waitFor } from '@testing-library/react'
+import AdvancedSearch from "../AdvancedSearch"
+import { QueryClient, QueryClientProvider } from "react-query";
+import { MemoryRouter, Routes, Route } from "react-router-dom";
+import userEvent from "@testing-library/user-event";
+
+const mockUseNavigate = jest.fn();
+jest.mock('react-router-dom', () => ({
+   ...jest.requireActual('react-router-dom') as any,
+  useNavigate: () => mockUseNavigate,
+}));
+
+const mockGetAllCategories = jest.fn()
+jest.mock('../../../api/categories', () => ({
+    getAllCategories: () => mockGetAllCategories(),
+}))
+
+const queryClient = new QueryClient()
+const user = userEvent.setup()
+
+const renderPage = (
+    queryClient: QueryClient,
+    collection = "example collection" 
+    ) => {
+        return render(
+            <QueryClientProvider client={queryClient}>
+                <MemoryRouter initialEntries={[`/collections/${collection}/artworks`]}>
+                    <Routes>
+                        <Route path="/collections/:collection/artworks" element={<AdvancedSearch collectionName={collection} />}/>
+                    </Routes>  
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+};
+
+const fetchedCategories = {
+    categories: [
+        "Tytuł",
+        "Tytuł.Podtytuł",
+        "Artyści",
+        "Rok"
+    ]
+}
+
+describe("AdvancedSearch tests", () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        queryClient.clear();
+    });
+
+    test("component renders correctly", async () => {        
```

Comment:
Zamiast "test" lepiej użyć "it" i pisać testy w konwencji `it("should render smth", () => {...}) `

Reply:
A i ten tst nie musi być `async`hroniczny